pkg/signals/signals.go

```diff
@@ -0,0 +1,89 @@
+// Package signals exposes signal handler.
+package signals
+
+import (
+	"log"
+	"os"
+	"os/exec"
+	"os/signal"
+)
+
+const (
+	// signalAbort
+	signalAbort signalType = 1
+	// signalTerminate
+	signalTerminate signalType = 2
+)
+
+type signalType int
+
+// Handler defines signal handler.
+type Handler struct {
+	// signalCh stores signals received from the system.
+	signalCh chan os.Signal
+	// signalType stores whether to abort gracefully or terminate.
+	signalType signalType
+}
+
+// NewSignalHandler creates a new Handler.
+func NewSignalHandler() *Handler {
+	// Different signals that we are listening to. For the time being putting
+	// only os.Interrupt, making a slice so that we can easily add more.
+	signals := []os.Signal{os.Interrupt}
+
+	signalCh := make(chan os.Signal, 1)
+	signal.Notify(signalCh, signals...)
+
+	return &Handler{
+		signalCh: signalCh,
+	}
+}
+
+// Register starts listening for the os interrupts.
+func (h *Handler) Register() {
+	go func() {
+		defer signal.Stop(h.signalCh)
+
+		for s := range h.signalCh {
+			if s == os.Interrupt {
+				if h.signalType == 0 {
+					h.signalType = signalAbort
+
+					log.Println("SIGINT Received, waiting for the graceful shutdown")
+
+					continue
+				}
+
+				log.Println("Second interrupt received, exiting immediately.")
+
+				h.signalType = signalTerminate
+
+				os.Exit(1) //nolint:gomnd
+			}
+		}
+	}()
+}
+
+// RegisterCmd starts listening for the os interrupts for the provided Cmd.
+// and passes the interrupt to the running process.
+func (h *Handler) RegisterCmd(cmd *exec.Cmd) {
+	go func() {
```

Comment:
Do we need a goroutine for each cmd? Maybe it would be easier to collect them and once we receive the signal, iterate?